tests/apollo/test_skvbc_diagnostics.py

```diff
@@ -0,0 +1,122 @@
+from util.bft import with_trio, with_bft_network, KEY_FILE_PREFIX, ConsensusPathType
+from util.skvbc_history_tracker import verify_linearizability
+from util import skvbc as kvbc
+from typing import TYPE_CHECKING, Sequence
+
+sys.path.append(str(Path(__file__).absolute().parent.parent.parent / "diagnostics"))
+import concord_ctl_logic as concord_ctl
+
+if TYPE_CHECKING:
+    from util.bft import BftTestNetwork
+    from util.skvbc_history_tracker import SkvbcTracker
+
+
+def replica_diagnostic_server_port(replica_id):
+    """
+    Returns a unique port for each replica
+    All the replicas run in the same container and therefore share an IP address
+    """
+    return 27500 + replica_id
+
+
+def start_replica_cmd(builddir, replica_id):
+    """
+    Return a command that starts an skvbc replica when passed to
+    subprocess.Popen.
+
+    Note each arguments is an element in a list.
+    """
+    statusTimerMilli = "500"
+    view_change_timeout_milli = "10000"
+    path = os.path.join(builddir, "tests", "simpleKVBC", "TesterReplica", "skvbc_replica")
+    return [path,
+            "-k", KEY_FILE_PREFIX,
+            "-i", str(replica_id),
+            "-s", statusTimerMilli,
+            "-v", view_change_timeout_milli,
+            "-e", str(True),
+            "--diagnostics-port", f"{replica_diagnostic_server_port(replica_id)}"
+            ]
+
+
+class SkvbcDiagnoticsTest(ApolloTest):
+    """
+    This class tests the diagnostic server in skvbc_replica
+    It uses the concord-ctl utility's functionality to communicate with the server
+    """
+    __test__ = False  # so that PyTest ignores this test scenario
+    # There is no need to rotate keys as it is unrelated to path transition and takes time
+    ROTATE_KEYS = False
+
+    @property
+    def hostname(self):
+        """
+        Hostname for diagnostics server
+        """
+        return 'localhost'
+
+    def run_ctl_command(self, command_args: Sequence[str], replica_id:int = 0):
+        return concord_ctl.run(self.hostname, replica_diagnostic_server_port(replica_id),
+                               command_args, print_output=False)
+
+    def validate_status_get_command(self, command_args: Sequence[str], replica_id:int = 0):
+        invalid_response = '*--STATUS_NOT_FOUND--*'
+        output = self.run_ctl_command(['status', 'get'] + list(command_args), replica_id)
+        assert invalid_response not in output
+        return output
+
+    def validate_status_describe_command(self, command_args: Sequence[str], replica_id:int = 0):
+        invalid_response = '*--DESCRIPTION_NOT_FOUND--*'
+        output = self.run_ctl_command(['status', 'describe'] + list(command_args), replica_id)
+        assert invalid_response not in output
+        return output
+
+    @with_trio
+    @parameterize(is_status_valid=[True, False])
+    @with_bft_network(start_replica_cmd, bft_configs=[BFTConfig(n=4)], rotate_keys=ROTATE_KEYS)
+    @verify_linearizability()
```

Comment:
you don't need linearizability checks for diagnostic status tests

Reply:
I do, since I need the skvbc object which needs a tracker...

Reply:
@WildFireFlum actually, the tracker is optional in the skvbc object:
https://github.com/vmware/concord-bft/blob/master/tests/apollo/test_skvbc.py#L64

This is to enable testing with/without linearizability checks.

Reply:
Changed, thank you.